crates/red_knot_python_semantic/src/types.rs

```diff
@@ -296,6 +297,42 @@ impl<'db> Type<'db> {
         }
     }
 
+    /// Return true if this type is assignable to type `other`.
+    #[allow(unused)]
+    pub(crate) fn is_assignable_to(self, db: &'db dyn Db, other: Type<'db>) -> bool {
+        if self.is_equivalent_to(db, other) {
+            return true;
+        }
```

Comment:
I would move the `is_equivalent` into the default branch on line 324 so that we can get a full exhaustiveness check. Except if calling `is_equivalent` first is good for perf

Reply:
I did it for performance reasons, so that the first thing we check is simple Eq of the types, before doing any more unnecessary work. (But I haven't done any benchmarking to verify that this makes a real difference.)